### PR TITLE
Use std's pattern matching functions instead of byte-wise comparisons.

### PR DESCRIPTION
**Pull request by adamreichold**

### CODE REVIEW

File: src/parse.rs [hunk 1]
```diff
@@ -459,8 +459,8 @@ fn parse(text: &str, opt: ParsingOptions) -> Result<Document, Error> {
     };
 
     // Trying to guess rough nodes and attributes amount.
-    let nodes_capacity = text.bytes().filter(|c| *c == b'<').count();
-    let attributes_capacity = text.bytes().filter(|c| *c == b'=').count();
+    let nodes_capacity = text.match_indices('<').count();
+    let attributes_capacity = text.match_indices('=').count();
```
Comment (tomjw64) [inline src/parse.rs:463]: Curious, why `match_indices` instead of `matches`?

**Comment by RazrFalcon** [comment]
Have you run the benchmarks? Is it faster?

File: src/parse.rs (hunk 1 above)
Comment (adamreichold) [reply src/parse.rs:463]: No reason besides looking through the methods and not seeing `matches`. Will change.

**Comment by adamreichold** [comment]
> Have you run the benchmarks? Is it faster?

Not yet. Will do, probably this evening.

**Comment by RazrFalcon** [comment]
Ok. I'm just confused by this change. Is it purely stylistic or actually has some impact on parsing.

**Comment by adamreichold** [comment]
So it is consistently slower:

```console
> cargo benchcmp master pattern
 name              master ns/iter  pattern ns/iter  diff ns/iter  diff %  speedup 
 large_roxmltree   3,261,437       4,048,171             786,734  24.12%   x 0.81 
 medium_roxmltree  769,474         855,463                85,989  11.18%   x 0.90 
 tiny_roxmltree    4,981           5,532                     551  11.06%   x 0.90 
```

which I was _not_ expecting. Thank you calling out the missing benchmarks.

(I had assumed that std would be clever enough to use libc's optimized implementations of `memchr` and/or `memmem` but that does not seem to be the case.)

So I definitely withdraw this. Sorry for the noise!

**Comment by RazrFalcon** [comment]
No problem! That's why we have benchmarks.

I remember testing this to get as best results as possible without relying on `memchr`, since I don't want an additional dependency.

Also, afair, the `memchr` crate uses SIMD only on x86. So on ARM it will use the same code as we do already.

**Pull request closed without merging**